hutch_python/epics_arch.py

```diff
@@ -156,33 +156,93 @@ def create_arch_file(experiment, level=None, hutch=None, path=None, dry_run=Fals
         print_dry_run(experiment)
 
 
-def pull_cds_data(exp, run):
+def pull_cds_items(exp):
+    logger.debug("in client")
+
     """
-    Gather all user objects from the CDS tab in the questionnaire.
-    Parse objects and separate them based on type.
-    Display them in the console via PrettyTable.
+    Gather all user obejcts from the CDS tab in the questionnaire.
+    Parse objects and sperate them based on type.
+    Display them in the console vie PrettyTable.
 
     Parameters
     ----------
     exp: ``str``
         The experiment's name e.g. xppx1003221
     run: ''str''
         The run number e.g. run21
-
+    ----------
     Outputs
     -------
         PrettyTable visualization of cds objects
+    -------
+
     """
-    # pull run data from questionnaire api, then take the data and sort it
-    # create PrettyTable instance and if the values from the run data contain
-    # pcdssetup then put them into a seperate dictionary as they are cds items
-    logger.debug('pull_cds_items(%s)', exp)
-    client = QuestionnaireClient()
-    logger.debug("in cds items, run numb:", str(run[1]))
-    runDetails_Dict = client.getProposalDetailsForRun(str(run[0]), str(run[1]))
+    """
+    pull run data from questionnaire api, then take the data and sort it
+    create Pretty Table instance and if the values from the run data contain pcdssetup
+    then put them into a seperate dictionary as they are cds items
+    """
+
+    logger.debug('pull_cds_items:', exp)
+    try:
+        client = QuestionnaireClient()
+    except GSSError:
+        print('No kerberos token found; rerun after running kinit')
+        sys.exit()
```

Comment:
You should avoid calling `sys.exit` in functions other than the main entry point, else someone who tries to import and use this elsewhere might have their whole program close unexpectedly, when maybe they just want to know that this failed and do something else- perhaps in a gui for example.

It should be sufficient to return here, or maybe re-raise the exception?

```suggestion
        return
```

```suggestion
        raise
```